File: Backend/src/main/java/com/example/traininfo/service/TrainService.java
package com.example.traininfo.service;

import java.util.List;

import com.example.traininfo.dto.TrainDTO;

import com.example.traininfo.entity.Train;

public interface TrainService {
	
	Train createTrain(TrainDTO trainDTO);

	Train updateTrain(Long id, TrainDTO trainDTO);

	boolean deleteTrain(Long id);

	Train getTrainById(Long id);

	Train getTrainByNumber(Integer number);

	List<Train> searchTrainsByName(String name);

	List<Train> searchTrainsBySeatsAvailable(int seatsAvailable);

	List<Train> getAllTrains();
}

File: Frontend/src/app/service/train.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Train } from '../model/train';

@Injectable({
  providedIn: 'root'
})
export class TrainService {
  private baseUrl = '';

  constructor(private http: HttpClient) { }

  createTrain(train: Train) {
    //write your logic here
    return null;
  }

  updateTrain(train: Train){
    //write your logic here
    return null;
  }

  deleteTrain(id: number){
    //write your logic here
    return null;
  }

  getTrainById(id: number){
    //write your logic here
    return null;
  }

  getAllTrains(){
    //write your logic here
    return null;
  }

  getTrainByNumber(number: number){
    //write your logic here
    return null;
  }

  getTrainsByName(name: string){
    //write your logic here
    return null;
  }

  getTrainsBySeatsAvailable(seatsAvailable: number){
    //write your logic here
    return null;
  }

}
